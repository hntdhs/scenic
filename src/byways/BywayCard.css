/* .byway-card {
    padding: 5em;
    border: groove 1px #DE9B72;
    text-align: center;
} */


.byway-card-name {
    color: green;
    font-family: fantasy;
    font-size: 2.5em;
}

.byway-card-image {
    width: 50%;
}

.byway-card-designation {
    color: green;
    font-size: 2em;
}