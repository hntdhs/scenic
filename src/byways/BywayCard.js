import React from "react";
import { Link } from "react-router-dom";
import "./BywayCard.css"

// name, nickname and image are state info, not byways - (name, state, length, designation, fees, image, description, geographic_features)
function BywayCard ({ name, designation, image, id }) {
    return (
        <span className="byway-card">
            <Link to={`/byways/${name}`} class="link-offset-2 link-underline link-underline-opacity-0">
                <h3 className="byway-card-name">{name}</h3>
                <img src={image} alt={name} className="byway-card-image"/>
                <h4 className="byway-card-designation">{designation}</h4>
            </Link>
        </span>
    );
}

export default BywayCard;