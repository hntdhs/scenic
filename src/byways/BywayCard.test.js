import React from "react";
import { render } from "@testing-library/react";
import BywayCard from "./BywayCard";
import { MemoryRouter } from "react-router";

it("matches snapshot", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <BywayCard
            name="Great River Road"
            image="https://fhwaapps.fhwa.dot.gov/bywaysp/uploads/asset_files/2279/82535_1-4Effigy%20Mounds%20National%20Monument%20Fire%20Point1_sq.jpg"
            designation="All-American Road"
            />
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});


it("matches snapshot", function () {
    const byway = { name: "byway", image: "imageURL", designation: "designation" }
    const { asFragment } = render(
        <MemoryRouter>
            <BywayCard byway={byway} />
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

// should there be comma after closing bracket of memory router? that's how it is in jobly
