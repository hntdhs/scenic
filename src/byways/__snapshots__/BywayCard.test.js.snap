// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <span
    class="byway-card"
  >
    <a
      class="link-offset-2 link-underline link-underline-opacity-0"
      href="/byways/Great River Road"
    >
      <h3
        class="byway-card-name"
      >
        Great River Road
      </h3>
      <img
        alt="Great River Road"
        class="byway-card-image"
        src="https://fhwaapps.fhwa.dot.gov/bywaysp/uploads/asset_files/2279/82535_1-4Effigy%20Mounds%20National%20Monument%20Fire%20Point1_sq.jpg"
      />
      <h4
        class="byway-card-designation"
      >
        All-American Road
      </h4>
    </a>
  </span>
</DocumentFragment>
`;

exports[`matches snapshot 2`] = `
<DocumentFragment>
  <span
    class="byway-card"
  >
    <a
      class="link-offset-2 link-underline link-underline-opacity-0"
      href="/byways/undefined"
    >
      <h3
        class="byway-card-name"
      />
      <img
        class="byway-card-image"
      />
      <h4
        class="byway-card-designation"
      />
    </a>
  </span>
</DocumentFragment>
`;
